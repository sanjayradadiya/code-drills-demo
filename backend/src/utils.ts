export const mockData = {
  Reports: [
    {
      ReportID: 'BalanceSheet',
      ReportName: 'Balance Sheet',
      ReportType: 'BalanceSheet',
      ReportTitles: [
        'Balance Sheet',
        'Demo Company (AU)',
        'As at 28 February 2018',
      ],
      ReportDate: '23 February 2018',
      UpdatedDateUTC: '/Date(1519358515899)/',
      Rows: [
        {
          RowType: 'Header',
          Cells: [
            { Value: '' },
            { Value: '28 Feb 2018' },
            { Value: '28 Feb 2017' },
          ],
        },
        {
          RowType: 'Section',
          Title: 'Assets',
        },
        {
          RowType: 'Section',
          Title: 'Bank',
          Rows: [
            {
              RowType: 'Row',
              Cells: [
                { Value: 'Business Bank Account' },
                { Value: '-2894.08' },
                { Value: '0.00' },
              ],
            },
            {
              RowType: 'Row',
              Cells: [
                { Value: 'Business Savings Account' },
                { Value: '6878.28' },
                { Value: '0.00' },
              ],
            },
            {
              RowType: 'SummaryRow',
              Cells: [
                { Value: 'Total Bank' },
                { Value: '3984.20' },
                { Value: '0.00' },
              ],
            },
          ],
        },
      ],
    },
  ],
};
