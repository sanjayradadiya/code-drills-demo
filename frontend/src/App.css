h1, h2, h3, h4 {
  padding: 2px;
  margin: 2px;
}
.container {
  text-align: center;
  margin: 20px 0;
}
.years {
  background: #0a0a23;
  color: #fff;
}

.years th {
  padding: 0.5rem;
  font-weight: bold;
  text-align: right;
  margin-right: 20px;
}

table-wrap {
  padding: 0 0.75rem 1.5rem 0.75rem;
}

table {
  border-collapse: collapse;
  border: 0;
  width: 900px;
  margin: 0 auto;
  position: relative;
  margin-top: 1rem;
}

table caption {
  color: #356eaf;
  font-size: 1.3em;
  font-weight: normal;
  position: absolute;
  top: -2.25rem;
  left: 0.5rem;
}

tbody td {
  width: 100vw;
  min-width: 4rem;
  max-width: 4rem;
}
tbody tr.title {
  width: 100%;
  background-image: linear-gradient(
    to bottom,
    #dfdfe2 1.845rem,
    white 1.845rem
  );
}
tbody tr.title th {
  text-align: left;
  padding: 0.5rem;
}
tbody td.space {
  width: 100vw;
}

tr.total {
  border-top: 2px solid #0a0a23;
  font-weight: bold;
  text-align: right;
}
tr.total td span.total {
  text-align: right;
  font-weight: bold;
}

tr.total th {
  text-align: left;
  padding: 0.5rem 0 0.25rem 0.5rem;
}

tr.total td {
  text-align: right;
  padding: 0.25rem;
}

tr.total td:last-of-type {
  padding-right: 0.5rem;
}

td.current {
  font-style: italic;
}

tr th.name {
  color: #000 !important;
  font-size: 1em !important;
  font-weight: bold !important;
}

tr.data td.no-data {
  text-align: center;
}

tr.data th {
  text-align: left;
  padding-top: 0.3rem;
  padding-left: 0.5rem;
}

tr.data th .description {
  display: block;
  font-weight: normal;
  font-style: italic;
  padding: 1rem 0 0.75rem;
  margin-right: -13.5rem;
}

tr.data td {
  vertical-align: top;
  padding: 0.3rem 0.25rem;
  text-align: right;
}
tr.data td.caption {
  vertical-align: top;
  padding: 0.3rem 0.25rem;
  text-align: left;
  margin-right: 20px;
}

tr.data td:last-of-type {
  padding-right: 0.5rem;
}
