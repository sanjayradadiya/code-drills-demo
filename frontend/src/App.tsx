import { useCallback, useEffect, useState } from 'react';
import './App.css';
import TableHeader from './components/TableHeader';
import TableBody from './components/TableBody';

const BalanceSheetTable = ({ data }: { data: any }) => {
  const rows = data?.Reports?.[0]?.Rows;
console.log('rows ==>', data)
  if (rows?.length === 0) {
    return null;
  }
  return (
    <>
      <table>
        <TableHeader rows={rows} />
        <TableBody rows={rows} />
      </table>
    </>
  );
};

export default function App() {
  const [data, setData] = useState([]);

  const handleBalanceSheetFetch = useCallback(async () => {
    const _data = await fetch('http://localhost:3005/BalanceSheet');
    console.log('_data', _data)
    const balanceSheet = await _data.json();
    console.log('balanceSheet', balanceSheet)
    setData(balanceSheet);
  }, []);

  useEffect(() => {
    handleBalanceSheetFetch();
  }, []);
  return <BalanceSheetTable data={data} />;
}
