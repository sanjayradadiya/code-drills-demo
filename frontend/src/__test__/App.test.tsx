import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import App from '../App';
import { mockData } from './utils';

describe('renders learn react link', () => {
  const {asFragment} = render(<App />);

  it('matches snapshot', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays fetched data', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockData),
      }),
    ) as jest.Mock;
    render(<App />);
    // Wait for the fake fetch to finish 
    expect(fetch).toHaveBeenCalledWith('http://localhost:3005/BalanceSheet');
    await waitFor(async() => {
      expect(await screen.findByText('28 Feb 2018')).toBeInTheDocument()
    })
    await waitFor(async() => {
      expect(await screen.findByText('Total Bank')).toBeInTheDocument()
    })
  });

});
