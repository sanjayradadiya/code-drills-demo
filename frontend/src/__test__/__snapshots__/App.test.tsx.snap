// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr
        class="years"
      >
        <td />
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;
